### DISCUSSION TITLE
Is it possible to use ref parameters in functions?

### DISCUSSION THREAD

**Discussion by philpax** [sebastienros/jint#1145] [Q&A] [answered]
Hi there,

I am in the process of binding jint to my application, and am enjoying the experience. I've used the CLR interop to expose ImGUI.NET, which has methods with signatures like this:
`public static bool Checkbox(string label, ref bool v)`

A cursory attempt to use this the naive way (`ImGui.Checkbox("Some Label", bool_variable)`) results in an overload resolution error: 
`No public methods with the specified arguments were found.`

My guess is that there's no way to make this work 'out of the box', given that JS doesn't have explicit-ref semantics. I'm seeking confirmation on that before I start writing wrappers to hide the `ref`s from the API 😅

**Comment by lahma** [comment, marked as answer]
Currently this is not possible, like you've discovered. It can become quite hard to support as under the hood the JsBoolean parameter is converted to `bool`. This would mean that Jint would need to track ref-passed parameters and then set the value back to given variable name the way it's after the function call. Probably not impossible, but can get quite hairy...

**Reply by philpax** [reply]
Yup, that makes sense - there's also not really much precedent for this on the JS side (it would be _weird_ to see `let val = true; sets_to_false(val); assert(val == false)`). If this was something that the project wanted to implement in future, it'd best be done with some kind of wrapper IMO, like:
```javascript
let val = jintRef(true);
sets_to_false(val);
assert(val == false);
```

That being said, appreciate the answer! I'll just have to expose a more JS-idiomatic interface for the functions I care about 😅
